diff --git a/node_modules/@SiLeBAT/fg43-ne-server/lib/index.js b/node_modules/@SiLeBAT/fg43-ne-server/lib/index.js
index 1234567..abcdefg 100644
--- a/node_modules/@SiLeBAT/fg43-ne-server/lib/index.js
+++ b/node_modules/@SiLeBAT/fg43-ne-server/lib/index.js
@@ -86,7 +86,17 @@
                         code: 2,
                         message: err.message
                     })
                         .end();
-                }
+                } else if (err.status) {
+                    next(err);
+                } else {
+                    app.get('logger').error(`Unhandled server error. error=${String(err)}`);
+                    res.status(500)
+                        .send({
+                        code: 1,
+                        message: 'Internal Server Error'
+                    })
+                        .end();
+                }
             });
             app.get('*', (req, res) => {
